x-pack/legacy/plugins/maps/public/layers/styles/vector/components/color/dynamic_color_form.js

```diff
@@ -40,21 +46,42 @@ export function DynamicColorForm({
   };
 
   const onFieldChange = async ({ field }) => {
-    const { name, origin, type } = field;
+    const { name, origin, type: fieldType } = field;
+    const defaultColorMapType = getDefaultColorMapType(fieldType);
     onDynamicStyleChange(styleProperty.getStyleName(), {
       ...styleOptions,
       field: { name, origin },
-      type: CATEGORICAL_DATA_TYPES.includes(type)
-        ? COLOR_MAP_TYPE.CATEGORICAL
-        : COLOR_MAP_TYPE.ORDINAL,
+      type: defaultColorMapType,
```

Comment:
Why did you move the logic into the function getDefaultColorMapType? Seems like this did not need to change at all.